File: common/ruby_common/excel_manipulate.rb
#! /usr/bin/ruby1.8
# -*- coding: utf-8 -*-
#
#	excel_manipulate.rb
#
#					Jul/31/2011
#
# -------------------------------------------------------------------
require 'spreadsheet'

# -------------------------------------------------------------------
def excel_row_read_proc (sheet,row,dict_aa)
	id = sheet[row,0]

	if (id != nil)
		name = sheet[row,1]
		population = sheet[row,2]
		date_mod = sheet[row,3]

		dict_aa = dict_append_proc(dict_aa,id,name,population,date_mod)
	end

	return dict_aa
end
# -------------------------------------------------------------------
def excel_read_proc (file_excel)
	dict_aa = {}
#
	xls = Spreadsheet.open(file_excel, 'rb')
	sheet = xls.worksheet(0)
	for row in 0..8 do
		dict_aa = excel_row_read_proc(sheet,row,dict_aa)
	end

	return	dict_aa
end
#
# -------------------------------------------------------------------
def excel_row_create_proc (sheet,row,key,value)
	sheet[row,0] = key
	sheet[row,1] = value['name']
	sheet[row,2] = value['population']
	sheet[row,3] = value['date_mod']
end
# -------------------------------------------------------------------
def excel_write_proc (excel_file,dict_aa)
	book = Spreadsheet::Workbook.new
	sheet = book.create_worksheet
#
	row = 0
	dict_aa.each {|key, value|
		excel_row_create_proc(sheet,row,key,value)
		row = row + 1

	}

	book.write(excel_file)
end
# -------------------------------------------------------------------

File: common/python_common/excel_manipulate.py
# -*- coding: utf-8 -*-
#
#	excel_manipulate.py
#
#						Sep/05/2014
#
# -------------------------------------------------------------------
import sys
import string
import xlrd
import xlwt

# -------------------------------------------------------------------
def excel_read_proc (excel_file):
	dict_aa = {} 
#
	book = xlrd.open_workbook(excel_file)
#
	sheet_1 = book.sheet_by_index(0)
#
	for row in range(sheet_1.nrows):
		for col in range(sheet_1.ncols):
			key = sheet_1.cell(row,0).value
			name = sheet_1.cell(row,1).value
			population = int (sheet_1.cell(row,2).value)
			date_mod = sheet_1.cell(row,3).value
			dict_unit = {'name': name, \
				'population':population,'date_mod':date_mod}
			dict_aa [key] = dict_unit
#
	return	dict_aa
#
# -------------------------------------------------------------------
def excel_data_create_proc (ws,row,key,unit):
	ws.write(row,0,key)
	ws.write(row,1,unit['name'])
	ws.write(row,2,unit['population'])
	ws.write(row,3,unit['date_mod'])
# -------------------------------------------------------------------
def excel_write_proc (excel_file,dict_aa):
	wb = xlwt.Workbook()
	ws = wb.add_sheet ('Cities')
#
	irow = 0
	for key in dict_aa.keys():
		excel_data_create_proc (ws,irow,key,dict_aa[key])
		irow += 1

	wb.save (excel_file)
# -------------------------------------------------------------------
